### ISSUE TITLE
npm start gives error

### ISSUE THREAD

**Issue by dvatsal** [react-community/create-react-native-app#565]
Please make our job easier by filling this template out to completion. If you're requesting a feature instead of reporting a bug, please feel free to skip the Environment and Reproducible Demo sections.

## Description

1-2 sentences describing the problem you're having or the feature you'd like to request
npm start does not work and gives an error message
## Expected Behavior
it should start the server for the app
What action did you perform, and what did you expect to happen?
I created a normal hello world application and tried 
## Observed Behavior
create-react-native-app HelloWorldVatsal
it created the app and loaded all modules etc and gave success message on the terminal
**After that did as below**

cd HelloWorldVatsal
npm start

What actually happened when you performed the above actions?

If there's an error message, please paste the *full terminal output and error message* in this code block:

```

> HelloWorldVatsal@0.1.0 start /home/mvix-vatsal/HelloWorldVatsal
> react-native-scripts start

14:52:58: Unable to start server
  See https://git.io/v5vcn for more information, either install watchman or run the following snippet:
    sudo sysctl -w fs.inotify.max_user_instances=1024
    sudo sysctl -w fs.inotify.max_user_watches=12288
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! HelloWorldVatsal@0.1.0 start: `react-native-scripts start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the HelloWorldVatsal@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mvix-vatsal/.npm/_logs/2018-02-19T09_22_58_504Z-debug.log

```

## Environment

Please run these commands in the project folder and fill in their results:

* `npm ls react-native-scripts`:npm ls react-native-scripts
* `npm ls react-native`:npm ls react-native
* `npm ls expo`:expo@25.0.0
* `node -v`:v9.3.0
* `npm -v`:4.6.1
* `yarn --version`:yarn is currently not installed
* `watchman version`:command not found

Also specify:

1. Operating system:4.13.0-32-generic #35~16.04.1-Ubuntu SMP Thu Jan 25 10:13:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
2. Phone/emulator/simulator & version:n/a

## Reproducible Demo

Please provide a minimized reproducible demonstration of the problem you're reporting.
create a new app using this script.
go to the app folder 
issue command 
npm start
Issues that come with minimal repro's are resolved much more quickly than issues where a maintainer has to reproduce themselves.

**Comment by andreasback** [comment]
Try running:

```
sudo sysctl -w fs.inotify.max_user_instances=1024
sudo sysctl -w fs.inotify.max_user_watches=12288
```
before

`npm start`

This will temporarily increase the number of files that the development server is able to monitor for changes.

This solution is specific for Ubuntu/Linux - the macOS equivalent is "kern.maxfiles" and "kern.maxfilesperproc" - there is a discussion about this problem [here](https://git.io/v5vcn)

**Comment by AdmiralRob** [comment]
react-native-scripts@1.11.1
react-native@0.52.0
expo@25.0.0
node -v:v8.0.0
npm -v:4.6.1
yarn --version: yarn is currently not installed
watchman version: command not found
Operating system: OS X Yosemite v10.10.5

I just installed react-native and got this exact same error. I ran sudo sysctl -w kern.maxfiles=5242880 and sudo sysctl -w kern.maxfilesperproc=524288 to no avail. They ran and appeared to do their job, but I still received the elifecycle error on running start.
